core/templates/dev/head/pages/exploration_editor/ExplorationMetadataFormDirective.js

```diff
@@ -0,0 +1,146 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the exploration metadata form.
+ */
+oppia.directive('explorationMetadataForm', [
+ 'UrlInterpolationService',
+ function(UrlInterpolationService) {
+   return {
+     restrict: 'E',
+     scope: {
+       settingsTab: '@',
```

Comment:
The naming here is not so good. It sounds like you are passing in the whole settings tab, and the exploration title itself, etc. -- where what you're trying to pass in instead is the context for the directive and the labels for the relevant fields. The names should reflect what exactly is being passed in.